code/datums/datumvars.dm

```diff
@@ -591,9 +592,37 @@ client
 		if(!istype(H))
 			usr << "This can only be used on instances of type /mob/living/carbon/human"
 			return
+			
+		var/confirm = alert("Are you sure you want to turn this mob into a skeleton?","Confirm Skeleton Transformation","Yes","No")
+		if(confirm != "Yes")
+			return
 
 		H.makeSkeleton()
 		href_list["datumrefresh"] = href_list["make_skeleton"]
+			
+	else if(href_list["offer_control"])
+		if(!check_rights(R_ADMIN))	return
+
+		var/mob/M = locate(href_list["offer_control"])
+		if(!istype(M))
+			usr << "This can only be used on instances of type /mob"
+			return
+		M << "Control of your mob has been offered to dead players."
+		log_admin("[key_name(usr)] has offered control of ([key_name(M)]) to ghosts.")
+		message_admins("[key_name_admin(usr)] has offered control of ([key_name_admin(M)]) to ghosts")
+		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", "Player Control", null, FALSE, 100)
+		var/mob/dead/observer/theghost = null
+
+		if(candidates.len)
+			theghost = pick(candidates)
+			M << "Your mob has been taken over by a ghost!"
+			message_admins("[key_name_admin(theghost)] has taken control of ([key_name_admin(M)])")
+			M.ghostize()

```

Comment:
ghostize() always makes a dead mob, client or not, if I recall correctly, and pointless dead mobs are indeed pointless